d2bs/kolbot/libs/bots/AutoBaal.js

```diff
@@ -101,25 +101,36 @@ function AutoBaal() {
 			} while (monster.getNext());
 		}
 
-		while (monList.length) {
-			monList.sort(Sort.units);
-			monster = copyUnit(monList[0]);
+		if (me.area === sdk.areas.ThroneofDestruction) {
+			[15116, 5026].distance > 10 && Pather.moveTo(15116, 5026);
+		}
+
+		let oldVal = Skill.usePvpRange;
+		Skill.usePvpRange = true;
+
+		try {
+			while (monList.length) {
+				monList.sort(Sort.units);
+				monster = copyUnit(monList[0]);
 
-			if (monster && monster.attackable) {
-				let index = monster.isSpecial ? 1 : 3;
+				if (monster && monster.attackable) {
+					let index = monster.isSpecial ? 1 : 3;
 
-				if (Attack.checkResist(monster, Attack.getSkillElement(Config.AttackSkill[index]))) {
-					if (Config.AttackSkill[index] > -1) {
-						ClassAttack.doCast(monster, Config.AttackSkill[index], Config.AttackSkill[index + 1]);
+					if (Attack.checkResist(monster, Attack.getSkillElement(Config.AttackSkill[index]))) {
+						if (Config.AttackSkill[index] > -1) {
+							ClassAttack.doCast(monster, Config.AttackSkill[index], Config.AttackSkill[index + 1]);
+						}
```

Comment:
What happens outside this if? Seems like no cast and no shift to next unit.

Reply:
Fixed, it was a pointless nested if